elog_zulip/__init__.py

```diff
@@ -312,39 +382,148 @@ def _replace_attachments_in_table(txt):
+                return f'[{match[1]}]({url})'
+
+            msg = re.sub(
+                fr'\[([^\]]*)\]\(({re.escape(current_elog_url)}/(\d+))\)',
+                get_zulip_entry_or_identity,
+                msg,
+                flags=re.I
+            )
+
+            return msg
+
+        def _fix_elog_links(msg):
+            return _fix_full_elog_links(_fix_relative_elog_links(msg))
+
         def _send_message(txt):
             txt = _replace_attachments_in_table(txt)
-            r = self._send_message(txt, topic, sender, date)
+            r = self._send_message(txt, topic, sender['email'], date)
             log.info(f'New publication: {self.entry_url(attributes)} - {r}')
+            return r
 
         # combine parts and send to zulip
         message = ''
+        first_zulip_message = None
         for part, part_images in parts:
             # TODO handle len(part) > maxchar
             if part_images:
                 part = _upload_embedded_images(part, part_images)
             if (len(message) + len(part)) > maxchar:
                 if message:
-                    _send_message(message)
+                    r = _send_message(_fix_elog_links(message))
+                    if first_zulip_message is None:
+                        first_zulip_message = r
                 message = part
             else:
                 message += part + os.linesep
         if message:
-            _send_message(message)
+            r = _send_message(_fix_elog_links(message))
+            if first_zulip_message is None:
+                first_zulip_message = r
+
+        if self.impersonate and is_user_new and self.can_create_users:
+            user_info = self._get_user_by_email(sender['email'])
+
+            if not user_info:
+                log.warning(f'User "{sender["email"]}" was not created for unknown reasons.')
+            else:
+                log.info(f'User "{sender["email"]}" was created and added to the user_map.')
+                self._users_map[sender['email']] = user_info
 
         # add entry to db
         data = {'entry_id': int(attributes["$@MID@$"]),
                 'entry_date': str(attributes['Date']),
                 'entry_author': str(attributes['Author'])}
         self.entry.insert(data, ['entry_id'])
 
-    def publish(self, id: int = None):
-        if id is not None:
-            self._publish(*self._read_entry(id))
-            return
+        self.elog_zulip_map.insert({
+            'entry_id': int(attributes["$@MID@$"]),
+            'entry_url': re.sub(r'(?<!:)//', '/', self.entry_url(attributes)).lower(),
+            'zulip_url': f'{self._zulip_url}/#narrow/channel/{self.stream}/topic/{topic}/near/{first_zulip_message["id"]}'
+        })
 
-        for content, attributes, attachments in self.new_entries():
-            self._publish(content, attributes, attachments)
+    def publish(self, ids: list[int] = []):
```

Comment:
You're breaking the current interface by changing the argument type (and name) for `ids`. Please add the logic to keep working with `int` as well as `list` of int.
It is also discouraged to have mutable objects as default argument values, I suggest to change it to None.

Reply:
yes, I use that quite often.
Whe you define the argument in the argument parser, you can chose your default value `default=None`

Reply:
What I meant is actually if you are using that function in any other project. If this wasn't the case maybe the interface can just be changed.

Reply:
When I pass a list of ids now I check whether the entry is in the db. Before my changes, when inserting a single message, the check wasn't there. Are you ok with the check happening even in the single id case or not?